api/api.go

```diff
@@ -375,6 +375,9 @@ func procDeleteNetwork(c libnetwork.NetworkController, vars map[string]string, b
 	if !errRsp.isOK() {
 		return nil, errRsp
 	}
+	if isReservedNetwork(nw.Name()) {
```

Comment:
It would be nice if checks like this one were managed by the backend in libnetwork. api pkg should not know about this, otherwise we are coupling the rest server with the backend.
But we do not have an API in libnetwork where to handle this check, given the delete network is handled directly by the network object. 
Actually, maybe we can put this logic in Network.Delete() and return a types.BadRequestf() if the net name is a reserved one.
Basically, the Network object itself will do the check on its name.
Can you please see if we can make the change that way. Thanks.